code/modules/telesci/quantumpad.dm

```diff
@@ -0,0 +1,197 @@
+
+/obj/machinery/power/quantumpad
+	name = "quantum pad"
+	desc = "A bluespace quantum-linked telepad used for teleporting objects to other quantum pads."
+	icon = 'icons/obj/telescience.dmi'
+	icon_state = "qpad-idle"
+	anchored = TRUE
+	use_power = 1
+	idle_power_usage = 200
+	active_power_usage = 5000
+	var/teleport_cooldown = 400 //30 seconds base due to base parts
+	var/teleport_speed = 50
+	var/last_teleport //to handle the cooldown
+	var/teleporting = 0 //if it's in the process of teleporting
+	var/power_efficiency = 1
+	var/obj/machinery/power/quantumpad/linked_pad
+
+	//mapping
+	var/static/list/mapped_quantum_pads = list()
+	var/map_pad_id = "" as text //what's my name
+	var/map_pad_link_id = "" as text //who's my friend
+
+/obj/item/weapon/circuitboard/telepad
+	name = T_BOARD("telepad")
+	build_path = /obj/machinery/power/quantumpad
+	board_type = "machine"
+	origin_tech = list(TECH_POWER = 6, TECH_ENGINEERING = 4, TECH_BLUESPACE = 4)
+	req_components = list(
+							/obj/item/weapon/stock_parts/subspace/crystal = 1,
+							/obj/item/weapon/stock_parts/capacitor = 2,
+							/obj/item/stack/cable_coil = 5,
+							/obj/item/weapon/stock_parts/console_screen = 1)
+
+/obj/machinery/power/quantumpad/New()
+	..()
+	component_parts = list()
```

Comment:
Fun fact, you can make all this component mess into build_default_parts(/obj/item/weapon/circuitboard/telepad)